sdks/python/apache_beam/runners/worker/statecache.py

```diff
@@ -148,47 +148,43 @@ class StateCache(object):
 
   def __init__(self, max_entries):
     _LOGGER.info('Creating state cache with size %s', max_entries)
-    self._cache = self.LRUCache(max_entries, (None, None))
+    self._missing = None
+    self._cache = self.LRUCache(max_entries, self._missing)
     self._lock = threading.RLock()
     self._metrics = Metrics()
 
   @Metrics.counter_hit_miss("get", "hit", "miss")
   def get(self, state_key, cache_token):
     assert cache_token and self.is_cache_enabled()
     with self._lock:
-      token, value = self._cache.get(state_key)
-    return value if token == cache_token else None
+      return self._cache.get((state_key, cache_token))
 
   @Metrics.counter("put")
   def put(self, state_key, cache_token, value):
     assert cache_token and self.is_cache_enabled()
     with self._lock:
-      return self._cache.put(state_key, (cache_token, value))
+      return self._cache.put((state_key, cache_token), value)
```

Comment:
The combined use of `(state_key, cache_token)` simplifies the lookup. Thanks for fixing this.